sklearn/model_selection/_search.py

```diff
@@ -784,7 +915,7 @@ def __init__(self, estimator, param_grid, scoring=None, fit_params=None,
             n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
             pre_dispatch=pre_dispatch, error_score=error_score)
         self.param_grid = param_grid
-        _check_param_grid(param_grid)
+        self.parameter_names_ = _check_param_grid_or_dist(param_grid)
```

Comment:
And we shouldn't really be doing validation, or setting such attributes, until `fit` except for backwards compatibility.